### PR TITLE
Set up CI/CD for Search-api-data production

### LINKED ISSUE #1778: Setup CI/CD for PR 'Setup CI/CD for PR Predicting and storing air quality of parishes'
**Is your feature request related to a problem? Please describe.**
- Setup CI/CD for PR #1592 

**Additional context**
- This is part of the predict microservice and thus, CI implementation should be done to the predict-microservice step in the workflow files. https://github.com/airqo-platform/AirQo-api/blob/0462823364297b5b42eac17ed0329032a1784944/.github/workflows/deploy-apis-to-staging.yml#L591
- The IaC for the CD changes will be made to the [predict helm chart](https://github.com/airqo-platform/AirQo-api/tree/staging/k8s/predict).
- The configmap to store the environment variables is `env-predict-places-air-quality` in both prod and stage

### PR DESCRIPTION
**Pull request by petermark23**
**_WHAT DOES THIS PR DO?_**
Sets up CI/CD for Search-api-data for production
**_WHAT ISSUES ARE RELATED TO THIS PR?_**
#1778 
- Jira cards
    - []()
- GitHub issues
    - Closes #1778 

**_HOW DO I TEST OUT THIS PR?_**

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

File: .github/workflows/deploy-apis-to-production.yml [hunk 1]
```diff
@@ -1750,8 +1741,9 @@ jobs:
         with:
           action: fjogeleit/yaml-update-action@main # Action to retry
           with: |
-            valueFile: "k8s/network-uptime/values-prod.yaml"
-            propertyPath: "image.tag"
+            valueFile: "k8s/predict/new_directory/values-prod.yaml"
+            
+            propertyPath: "image.new_property"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1746]: Hi @petermark23 
What is the `new_property` value?
Could you be meaning `tag`?

File: k8s/predict/values-prod.yaml [hunk 2]
```diff
@@ -41,6 +42,18 @@ jobs:
...
         memory: 4000Mi
+  predictPlaces:
+    name:  airqo-predict-places-air-quality
+    configmap: env-predict-places-air-quality-production
+    schedule: 30 * * * *
+    suspend: true
+    resources:
+  limits:
+        cpu: 8000m
+        memory: 16000Mi
```
Comment (123MwanjeMike) [inline k8s/predict/values-prod.yaml:53]: Could we also correct this indenting?

File: .github/workflows/deploy-apis-to-production.yml [hunk 3]
```diff
@@ -1661,6 +1661,13 @@ jobs:
           target: production
           tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-train-job:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-train-job:latest
 
+      - name: Build and push predict places image
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          context: src/predict/jobs/new_directory
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1668]: what is the `new_directory`?
Could we specify the exact directory.

File: .github/workflows/deploy-apis-to-production.yml [hunk 4]
```diff
@@ -1661,6 +1661,13 @@ jobs:
...
+          tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-predict-places-air-quality:latest-update
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1669]: Let's use `latest` tag and not `latest-update`

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]
Thanks @petermark23 
I left a few comments

**Comment by Psalmz777** [comment]
@123MwanjeMike looks like Peter made changes ,I have made a few correction ,please review again

File: .github/workflows/deploy-apis-to-production.yml [hunk 5]
```diff
@@ -1673,24 +1680,26 @@ jobs:
             token: ${{ secrets.YAML_UPDATER_TOKEN }}
             message: "Update predict production image tag to ${{ needs.image-tag.outputs.build_id }}"
 
+<<<<<<< HEAD
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1683]: Let’s have this line removed

File: .github/workflows/deploy-apis-to-production.yml [hunk 6]
```diff
@@ -1661,6 +1661,13 @@ jobs:
           target: production
           tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-train-job:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-train-job:latest
 
+      - name: Build and push predict places image
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          context: src/predict/jobs/predict_places_air_quality
+          tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-predict-places-air-quality:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/stage-airqo-predict-places-air-quality:latest
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1669]: Why are we tagging this production image with the `stage` word in the tag name?

File: .github/workflows/deploy-apis-to-production.yml [hunk 7]
```diff
@@ -1673,24 +1680,26 @@ jobs:
...
 
+<<<<<<< HEAD
       - name: Build and push predict places image
         uses: docker/build-push-action@v3
         with:
           push: true
           context: src/predict/jobs/predict_places_air_quality
-          tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-predict-places-air-quality:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-predict-places-air-quality:latest
-
+          tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-predict-places-air-quality:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/stage-airqo-predict-places-air-quality:latest
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1689]: Can we remove the `stage` word in this production image tag name?

File: .github/workflows/deploy-apis-to-production.yml [hunk 8]
```diff
@@ -1673,24 +1680,26 @@ jobs:
...
       - name: Update corresponding helm values file(with retry)
         uses: Wandalen/wretry.action@master
         with:
           action: fjogeleit/yaml-update-action@main
           with: |
-            valueFile: "k8s/predict/predict-places-air-quality/values-prod.yaml"
+            valueFile: "k8s/predict/new_directory/values-prod.yaml"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1695]: I still see an unexpected `new_directory` in the path. Could it still be an oversight?

File: .github/workflows/deploy-apis-to-production.yml [hunk 9]
```diff
@@ -1673,24 +1680,26 @@ jobs:
...
             propertyPath: "image.tag"
             value: ${{ needs.image-tag.outputs.build_id }}
             branch: ${{ env.DEPLOY_BRANCH }}
             message: "Update predict places air quality staging image tag to ${{ needs.image-tag.outputs.build_id }}"
 
+=======
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1701]: Can we remove these conflict resolution lines 1701 and 1702

File: k8s/predict/values-prod.yaml (hunk 2 above)
Comment (123MwanjeMike) [reply k8s/predict/values-prod.yaml:53]: Thanks

File: .github/workflows/deploy-apis-to-production.yml [hunk 10]
```diff
@@ -1750,7 +1761,8 @@ jobs:
         with:
           action: fjogeleit/yaml-update-action@main # Action to retry
           with: |
-            valueFile: "k8s/network-uptime/values-prod.yaml"
+            valueFile: "k8s/predict/new_directory/values-prod.yaml"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-apis-to-production.yml:1764]: Let’s remove `new_directory` from the path

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]
Thanks @petermark23 and @Psalmz777 
I’ve left some more comments on where I we can improve.

**Review by Psalmz777 — APPROVED** [review]

**Pull request merged**